### ISSUE TITLE
Excessive memory usage for one particular image

### ISSUE THREAD

**Issue by ringw** [tensorflow/moonlight#15]
I identified a smallish (2500x3200) image ([source](http://imslp.org/wiki/Piano_Trio_(Ravel,_Maurice)#IMSLP01667)) where we exceed 13GB RAM. Larger images may use excessive RAM, but this is unusual for an image of this size. Maybe because the staves are especially small (staffline distance of 11), the scale-normalized staves are unusually large.

![imslp01667-000](https://user-images.githubusercontent.com/1391644/39961733-b29517e2-560a-11e8-859e-c9fe29aaf0ac.png)

**Comment by ringw** [comment]
Memory usage seems to be due to the KNN classifier, which computes the distance between each patch and 10K examples. Smaller staves are actually scaled up and take up more space than staves that are originally taller. The fix will be a simple neural network classifier model.
